File: hw/chavez/u3/hw27Observer/JavaScript/model/IBM.js
import Stock from "./Stock.js";

class IBM extends Stock {

    constructor(symbol, price) {
        super();
        this.symbol = symbol;
        this.price = price;
    }

    getSymbol() {
        return this.symbol;
    }

    setSymbol(symbol) {
        this.symbol = symbol;
        this.notifyObservers(symbol);
    }

    getPrice() {
        return this.price;
    }

    setPrice(price) {
        this.price = price;
        this.notifyObservers(price);
    }
}

export default IBM;

File: hw/vizcaino/u3/hw27Observer/hw27ObserverJava/Observer/src/main/java/ec/edu/espe/observer/IBM.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.observer;

/**
 *
 * @author Adrian Vizcaino <The-Softwarriors at ESPE>
 */
public class IBM extends Stock {
    public IBM(String symbol, double price) {
        this.symbol = symbol;
        this.price = price;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
        notifyObservers(price);
    }

    public String getSymbol() {
        return symbol;
    }

    public void setSymbol(String symbol) {
        this.symbol = symbol;
        notifyObservers(symbol);
    }
}
